<a href="https://colab.research.google.com/github/minsaee/python_demo/blob/master/101_python_%EC%A0%9C%EC%96%B4%EB%AC%B8_%EB%AC%B8%EC%A0%9C.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

제어문이나 comprehension을 이용한 문제풀이


[문제1]  
리스트에 저장된 문자열 중에서 길이가 5인 것들만 리스트 형태로 출력하도록 구현하세요.  

a= ['alpha', 'bravo', 'charlie', 'delta', 'echo', 'foxtrot', 'golf', 'hotel', 'india']  

[출력결과]     
['alpha', 'bravo', 'delta', 'hotel', 'india']

In [42]:
# 길이가 5인 문자열 추출
# 리스트형태로 출력

a= ['alpha', 'bravo', 'charlie', 'delta', 'echo', 'foxtrot', 'golf', 'hotel', 'india']

len_fi = []
for len_five in range(len(a)):
  if len(a[len_five]) == 5:
    len_fi.append(a[len_five])
print(len_fi)

range(0, 9)
True
['alpha', 'bravo', 'delta', 'hotel', 'india']


In [18]:
a= ['alpha', 'bravo', 'charlie', 'delta', 'echo', 'foxtrot', 'golf', 'hotel', 'india']

len_fi = [a[len_five] for len_five in range(len(a)) if len(a[len_five]) == 5]
print((len_fi))

['alpha', 'bravo', 'delta', 'hotel', 'india']


[문제2]    
가로 크기를 저장한 리스트를 활용해서 가로 크기가 불규칙한  리스트를 생성하도록 프로그램을 구현하세요.

a = [3, 1,3, 2, 5]  


[출력결과]  
[[0, 0, 0], [0], [0, 0, 0], [0, 0], [0, 0, 0, 0, 0]]


In [31]:
# 가로크기를 저장한 리스트 활용
# 가로크기가 불규칙한 리스트 생성

a = [3, 1, 3, 2, 5]    # 가로 크기를 저장한 리스트
b = []    # 빈 리스트 생성 > 바깥쪽 리스트

for i in a:      # 가로 크기를 저장한 리스트로 반복
    line = []    # 안쪽 리스트로 사용할 빈 리스트 생성
    for j in range(i):    # 리스트 a에 저장된 가로 크기만큼 반복
        line.append(0)
    b.append(line)        # 리스트 b에 안쪽 리스트를 추가

print(b)

[[0, 0, 0], [0], [0, 0, 0], [0, 0], [0, 0, 0, 0, 0]]


In [38]:
a = [3, 1, 3, 2, 5]
line = []
b = [line.append(0) for i in a for j in range(i)]
print(b)

[None, None, None, None, None, None, None, None, None, None, None, None, None, None]


[문제3]    
포트폴리오의 총 비용을 계산하는 프로그램을 구현하세요.   

portfolio = [{'name':'AA', 'shares':100, 'price':32.20},   
{'name':'IBM', 'shares':50, 'price':91.10},    
{'name':'CAT', 'shares':150, 'price':83.44},  
{'name':'MSFT', 'shares':200, 'price':51.23},  
{'name':'GE', 'shares':95, 'price':40.37},  
{'name':'MSFT', 'shares':50, 'price':65.10},  
{'name':'IBM', 'shares':100, 'price':70.44} ]       

[출력결과]  
44671.15

In [61]:
# 포트폴리오의 총 비용 구하기
portfolio = [{'name':'AA', 'shares':100, 'price':32.20},
{'name':'IBM', 'shares':50, 'price':91.10},
{'name':'CAT', 'shares':150, 'price':83.44},
{'name':'MSFT', 'shares':200, 'price':51.23},
{'name':'GE', 'shares':95, 'price':40.37},
{'name':'MSFT', 'shares':50, 'price':65.10},
{'name':'IBM', 'shares':100, 'price':70.44} ]

#name -> 7개 ( shares * price ) + all

for n,s in enumerate(portfolio):
  print(n,s)

s = []
cnt = ['shares']
prod = ['price']

0 {'name': 'AA', 'shares': 100, 'price': 32.2}
1 {'name': 'IBM', 'shares': 50, 'price': 91.1}
2 {'name': 'CAT', 'shares': 150, 'price': 83.44}
3 {'name': 'MSFT', 'shares': 200, 'price': 51.23}
4 {'name': 'GE', 'shares': 95, 'price': 40.37}
5 {'name': 'MSFT', 'shares': 50, 'price': 65.1}
6 {'name': 'IBM', 'shares': 100, 'price': 70.44}


[문제4]  
포트폴리오에서 100주 이상 보유한 종목의 리스트를 출력하는 프로그램을 구현하세요.    
portfolio = [{'name':'AA', 'shares':100, 'price':32.20},{'name':'IBM', 'shares':50, 'price':91.10},{'name':'CAT', 'shares':150, 'price':83.44},{'name':'MSFT', 'shares':200, 'price':51.23},{'name':'GE', 'shares':95, 'price':40.37},{'name':'MSFT', 'shares':50, 'price':65.10},{'name':'IBM', 'shares':100, 'price':70.44} ]  


[출력결과]    
[{'name': 'AA', 'shares': 100, 'price': 32.2},
 {'name': 'CAT', 'shares': 150, 'price': 83.44},
 {'name': 'MSFT', 'shares': 200, 'price': 51.23},
 {'name': 'IBM', 'shares': 100, 'price': 70.44}]


[문제5]

보유한 MSFT와 IBM주식 전체 리스트를 출력하는 프로그램을 구현하세요.  

portfolio = [{'name':'AA', 'shares':100, 'price':32.20},{'name':'IBM', 'shares':50, 'price':91.10},{'name':'CAT', 'shares':150, 'price':83.44},{'name':'MSFT', 'shares':200, 'price':51.23},{'name':'GE', 'shares':95, 'price':40.37},{'name':'MSFT', 'shares':50, 'price':65.10},{'name':'IBM', 'shares':100, 'price':70.44} ]  

[출력결과]   
[{'name': 'IBM', 'price': 91.1, 'shares': 50},
 {'name': 'MSFT', 'price': 51.23, 'shares': 200},
 {'name': 'MSFT', 'price': 65.1, 'shares': 50},
 {'name': 'IBM', 'price': 70.44, 'shares': 100}]  



[문제6]
보유 종목 중 가격이 10,000달러가 넘는 것의 리스트를 구현하는 프로그램을 구현하세요.  
portfolio = [{'name':'AA', 'shares':100, 'price':32.20},{'name':'IBM', 'shares':50, 'price':91.10},{'name':'CAT', 'shares':150, 'price':83.44},{'name':'MSFT', 'shares':200, 'price':51.23},{'name':'GE', 'shares':95, 'price':40.37},{'name':'MSFT', 'shares':50, 'price':65.10},{'name':'IBM', 'shares':100, 'price':70.44} ]

[출력결과]    
[{'name': 'CAT', 'price': 83.44, 'shares': 150},
 {'name': 'MSFT', 'price': 51.23, 'shares': 200}]

[문제7]   
portfolio에서 얻은 name과 shares를 가지고 튜플의 리스트 (name, shares)를 만드는 프로그램을 구현하세요.  
portfolio = [{'name':'AA', 'shares':100, 'price':32.20},{'name':'IBM', 'shares':50, 'price':91.10},{'name':'CAT', 'shares':150, 'price':83.44},{'name':'MSFT', 'shares':200, 'price':51.23},{'name':'GE', 'shares':95, 'price':40.37},{'name':'MSFT', 'shares':50, 'price':65.10},{'name':'IBM', 'shares':100, 'price':70.44} ]

[출력결과]   
[('AA', 100),
 ('IBM', 50),
 ('CAT', 150),
 ('MSFT', 200),
 ('GE', 95),
 ('MSFT', 50),
 ('IBM', 100)]



[문제8]  

리스트에서 int만 추출하는 프로그램을 구현하세요.  
oldlist = [1, 2, 'A', False, 3]

[출력결과]   
[1, 4, 9]  


[문제9]   
1부터 10까지의 홀수, 짝수로 출력하는 프로그램을 구현하세요.

[출력결과]   
{1: '홀수',
 2: '짝수',
 3: '홀수',
 4: '짝수',
 5: '홀수',
 6: '짝수',
 7: '홀수',
 8: '짝수',
 9: '홀수',
 10: '짝수'}
